src/dsp_tools/commands/xmlupload/make_rdf_graph/make_values.py

```diff
@@ -51,157 +69,142 @@ def make_values(properties: list[XMLProperty], res_bnode: BNode, lookups: Lookup
-        single_prop_graph, last_prop_name = _make_one_prop_graph(prop=prop, res_bnode=res_bnode, lookups=lookups)
+    for prop in values:
+        single_prop_graph, last_prop_name = _make_one_prop_graph(val=prop, res_bnode=res_bnode, iri_lookup=lookups)
         properties_graph += single_prop_graph
 
     return properties_graph, last_prop_name
 
 
-def _make_one_prop_graph(prop: XMLProperty, res_bnode: BNode, lookups: Lookups) -> tuple[Graph, URIRef]:
-    prop_name = get_absolute_iri(prop.name, lookups.namespaces)
-    match prop.valtype:
-        case "boolean" | "color" | "decimal" | "geometry" | "geoname" | "integer" | "time" | "uri" as val_type:
-            literal_info = RDF_PROP_TYPE_MAPPER[val_type]
-            transformer = RDF_LITERAL_TRANSFORMER_MAPPER[val_type]
-            properties_graph = _make_simple_prop_graph(
-                prop=prop,
+def _make_one_prop_graph(val: IntermediaryValue, res_bnode: BNode, iri_lookup: IRILookup) -> tuple[Graph, URIRef]:
+    match val:
+        case (
+            IntermediaryBoolean()
+            | IntermediaryColor()
+            | IntermediaryDecimal()
+            | IntermediaryGeometry()
+            | IntermediaryGeoname()
+            | IntermediaryInt()
+            | IntermediaryTime()
+            | IntermediaryUri()
+            | IntermediarySimpleText() as val_type
+        ):
+            literal_info = RDF_LITERAL_PROP_TYPE_MAPPER[type(val_type)]
+            properties_graph = _make_value_graph_with_literal_object(
+                val=val,
                 res_bn=res_bnode,
-                prop_name=prop_name,
                 prop_type_info=literal_info,
-                transformer=transformer,
-                permissions_lookup=lookups.permissions,
-            )
-        case "list":
-            properties_graph = _make_list_prop_graph(
-                prop=prop,
-                res_bn=res_bnode,
-                prop_name=prop_name,
-                permissions_lookup=lookups.permissions,
-                listnode_lookup=lookups.listnodes,
             )
-        case "resptr":
-            prop_name = get_absolute_iri(f"{prop.name}Value", lookups.namespaces)
-            properties_graph = _make_link_prop_graph(
-                prop=prop,
+        case IntermediaryList():
+            properties_graph = _make_list_value_graph(val=val, res_bn=res_bnode, prop_type_info=LIST_PROP_TYPE_INFO)
+        case IntermediaryLink():
+            properties_graph = _make_link_value_graph(
+                val=val,
                 res_bn=res_bnode,
-                prop_name=prop_name,
-                permissions_lookup=lookups.permissions,
-                iri_resolver=lookups.id_to_iri,
+                prop_type_info=LINK_PROP_TYPE_INFO,
+                iri_resolver=iri_lookup.id_to_iri,
             )
-        case "text":
-            properties_graph = _make_text_prop_graph(
-                prop=prop,
+        case IntermediaryRichtext():
```

Comment:
Even with the improved transformation, it is not possible for some values to be constructed through a generic route.